library/built-in/_prefabs/prefab_cnet.ts

```diff
@@ -98,108 +98,101 @@ export type Cnet_return = {
     ckpt_return: _MODEL
 }
 
-export const run_cnet = async (opts: OutputFor<typeof ui_cnet>, cnet_args: Cnet_args) => {
+export const run_cnet = async (opts: OutputFor<typeof ui_cnet>, ctx: Cnet_args) => {
     const run = getCurrentRun()
-    const graph = run.nodes
-    // var positive = cnet_args.positive
-    // var negative = cnet_args.negative
-    // CNET APPLY
     const cnetList = opts?.controlNetList
-    let ckpt_return = cnet_args.ckptPos
-    let cnet_positive = cnet_args.positive
-    let cnet_negative = cnet_args.negative
+    let args: Cnet_args = { ...ctx }
 
     if (cnetList) {
         for (const cnetImage of cnetList) {
             let image: IMAGE = (await run.loadImageAnswer(cnetImage.image))._IMAGE
-            let cnet_name: Enum_ControlNetLoader_control_net_name
 
             if (cnetImage.cnets.IPAdapter) {
                 // IPAdapter APPLY ===========================================================
-                const ip_adapter_result = run_cnet_IPAdapter(cnetImage.cnets.IPAdapter, cnet_args, image)
-                ckpt_return = (await ip_adapter_result).ip_adapted_model
+                const ip_adapter_result = run_cnet_IPAdapter(cnetImage.cnets.IPAdapter, ctx, image)
+                args.ckptPos = ip_adapter_result.ip_adapted_model
             } else {
-                let strength: number | undefined
                 // CANNY ===========================================================
-                if (cnetImage.cnets.Canny) {
-                    const cnet_return_canny = await run_cnet_canny(cnetImage.cnets.Canny, cnet_args, image)
-                    strength = cnetImage.cnets.Canny.strength
-                    image = cnet_return_canny.image
-                    cnet_name = cnet_return_canny.cnet_name
+                const { Canny, Depth, Normal, Lineart, OpenPose, Scribble, SoftEdge, Tile } = cnetImage.cnets
+                if (Canny) {
+                    const y = run_cnet_canny(Canny, ctx, image)
+                    _apply_cnet(args, Canny.strength, y.image, y.cnet_name)
                 }
                 // POSE ===========================================================
-                else if (cnetImage.cnets.OpenPose) {
-                    const cnet_return_openPose = await run_cnet_openPose(cnetImage.cnets.OpenPose, cnet_args, image)
-                    strength = cnetImage.cnets.OpenPose.strength
-                    image = cnet_return_openPose.image
-                    cnet_name = cnet_return_openPose.cnet_name
+                if (OpenPose) {
+                    const y = run_cnet_openPose(OpenPose, ctx, image)
+                    _apply_cnet(args, OpenPose.strength, y.image, y.cnet_name)
                 }
                 // DEPTH ===========================================================
-                else if (cnetImage.cnets.Depth) {
```

Comment:
the `else` here was causing only one of the cnet to be taken in account, due to how choices widget works